File: iteration1/Person.java
package iteration1;

public abstract class Person {
    private int ID;
    private String name;
    private String surname;
    private Department department;
    private String userName;
    private String password;

    public Person(int ID, String name, String surname, String userName, String password) {
        this.ID = ID;
        this.name = name;
        this.surname = surname;
        this.userName = userName;
        this.password = password;
    }
    abstract boolean login(String userName, String password);
    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public Department getDepartment() {
        return department;
    }

    public void setDepartment(Department department) {
        this.department = department;
    }

    public String getName() {
        return name;
    }

    public String getSurname() {
        return surname;
    }

    public int getID() {
        return ID;
    }
}

File: iteration3/Person.py
from abc import ABC, abstractmethod


class Person(ABC):
    def __init__(self, ID, name, surname, userName, password, department=None):
        self.__ID = ID
        self.__name = name
        self.__surname = surname
        self.__userName = userName
        self.__password = password
        self.__department = department

    @abstractmethod
    def login(self, userName, password):
        pass

    def getUserName(self):
        return self.__userName

    def setUserName(self, userName):
        self.__userName = userName

    def getPassword(self):
        return self.__password

    def setPassword(self, password):
        self.__password = password

    def getDepartment(self):
        return self.__department

    def setDepartment(self, department):
        self.__department = department

    def getName(self):
        return self.__name

    def getSurname(self):
        return self.__surname

    def getID(self):
        return self.__ID
